### ISSUE TITLE
App crashes when runInitSequence method from ScienceLab.java is called

### ISSUE THREAD

**Issue by akarshan96** [fossasia/pslab-android#175]
**Actual Behaviour**

Calling runInitSequence method from ScienceLab.java is causing the app to crash.

**Expected Behaviour**

 The app shouldn't crash when runInitSequence is called.

**Steps to reproduce it**

Create an object of ScienceLab and call runInitSequence method.

**LogCat for the issue**

```
E/AndroidRuntime: FATAL EXCEPTION: main
                  Process: org.fossasia.pslab, PID: 32499
                  java.lang.RuntimeException: Unable to start activity ComponentInfo{org.fossasia.pslab/org.fossasia.pslab.activity.OscilloscopeActivity}: java.util.NoSuchElementException
                      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2423)
                      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2483)
                      at android.app.ActivityThread.access$900(ActivityThread.java:153)
                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1349)
                      at android.os.Handler.dispatchMessage(Handler.java:102)
                      at android.os.Looper.loop(Looper.java:148)
                      at android.app.ActivityThread.main(ActivityThread.java:5438)
                      at java.lang.reflect.Method.invoke(Native Method)
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:738)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:628)
                   Caused by: java.util.NoSuchElementException
                      at java.util.ArrayList$ArrayListIterator.next(ArrayList.java:576)
                      at java.util.Collections.min(Collections.java:1654)
                      at org.fossasia.pslab.communication.ScienceLab.loadTable(ScienceLab.java:2591)
                      at org.fossasia.pslab.communication.ScienceLab.loadEquation(ScienceLab.java:2556)
                      at org.fossasia.pslab.communication.ScienceLab.runInitSequence(ScienceLab.java:143)
                      at org.fossasia.pslab.activity.OscilloscopeActivity.onCreate(OscilloscopeActivity.java:144)
                      at android.app.Activity.performCreate(Activity.java:6303)
                      at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1108)
                      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2376)
                      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2483) 
                      at android.app.ActivityThread.access$900(ActivityThread.java:153) 
                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1349) 
                      at android.os.Handler.dispatchMessage(Handler.java:102) 
                      at android.os.Looper.loop(Looper.java:148) 
                      at android.app.ActivityThread.main(ActivityThread.java:5438) 
                      at java.lang.reflect.Method.invoke(Native Method) 
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:738) 
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:628) 
```

**Screenshots of the issue**

none

**Would you like to work on the issue?**

Yes

**Comment by viveksb007** [comment]
@akarshan96 could you also push your code changes to some branch of your repo. Would help in understanding the issue better.

**Comment by akarshan96** [comment]
@viveksb007, the issue is solved. `y.clear()` [here](https://github.com/fossasia/pslab-android/blob/master/app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java#L2584) is also clearing the `points` ArrayList. Let me run some more tests. 
Thanks

**Comment by akarshan96** [comment]
Fixed in #180
